Mathlib/ModelTheory/LanguageMap.lean

```diff
@@ -219,9 +220,12 @@ noncomputable def defaultExpansion (ϕ : L →ᴸ L')
 /-- A language homomorphism is an expansion on a structure if it commutes with the interpretation of
 all symbols on that structure. -/
 class IsExpansionOn (M : Type*) [L.Structure M] [L'.Structure M] : Prop where
-  map_onFunction : ∀ {n} (f : L.Functions n) (x : Fin n → M), funMap (ϕ.onFunction f) x = funMap f x
-  map_onRelation : ∀ {n} (R : L.Relations n) (x : Fin n → M),
-    RelMap (ϕ.onRelation R) x = RelMap R x
+  map_onFunction :
+    ∀ {n} (f : L.Functions n) (x : Fin n → M), funMap (ϕ.onFunction f) x = funMap f x :=
+      by exact fun {n} => (IsRelational.empty_functions _).elim
+  map_onRelation :
+    ∀ {n} (R : L.Relations n) (x : Fin n → M), RelMap (ϕ.onRelation R) x = RelMap R x :=
+      by exact fun {n} => (IsAlgebraic.empty_relations _).elim
```

Comment:
```suggestion
  map_onFunction :
    ∀ {n} (f : L.Functions n) (x : Fin n → M), funMap (ϕ.onFunction f) x = funMap f x := by
      exact fun {n} => (IsRelational.empty_functions _).elim
  map_onRelation :
    ∀ {n} (R : L.Relations n) (x : Fin n → M), RelMap (ϕ.onRelation R) x = RelMap R x := by
      exact fun {n} => (IsAlgebraic.empty_relations _).elim
```
also do you need the `by exact` at all? Are they an elaboration trick?

Reply:
It doesn't work when I don't use the `by exact`.
If I just try to use `IsAlgebraic`, say, to define something, it will (rightly) complain that there is no `IsAlgebraic` instance. If it's wrapped in `by exact`, it's willing to try and see whether there is an `IsAlgebraic` instance around.

Reply:
Okay, so they are an elaboration trick